src/modules/select/wallets/keepkey/index.ts

```diff
@@ -0,0 +1,389 @@
+        .catch((err: any) => callback(err, null))
+    },
+    signTransaction: (transactionData: any, callback: any) => {
+      signTransaction(transactionData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    processMessage: (messageData: any, callback: any) => {
+      signMessage(messageData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    processPersonalMessage: (messageData: any, callback: any) => {
+      signMessage(messageData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    signMessage: (messageData: any, callback: any) => {
+      signMessage(messageData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    signPersonalMessage: (messageData: any, callback: any) => {
+      signMessage(messageData)
+        .then((res: string) => callback(null, res))
+        .catch(err => callback(err, null))
+    },
+    rpcUrl
+  })
+
+  keyring.on(['*', '*', Events.DISCONNECT], async () => {
+    resetWalletState({ disconnected: true, walletName: WALLET_NAME })
+  })
+
+  // If the wallet asks for a PIN, open the PIN modal
+  keyring.on(['*', '*', Events.PIN_REQUEST], () => {
+    renderPinModal(keepKeyWallet)
+  })
+
+  provider.setPath = setPath
+  provider.dPath = dPath
+  provider.enable = enable
+  provider.setPrimaryAccount = setPrimaryAccount
+  provider.getPrimaryAddress = getPrimaryAddress
+  provider.getAccounts = getAccounts
+  provider.getMoreAccounts = getMoreAccounts
+  provider.getBalance = getBalance
+  provider.getBalances = getBalances
+  provider.send = provider.sendAsync
+  provider.disconnect = disconnect
+  provider.isCustomPath = isCustomPath
+
+  async function enable(): Promise<Array<string | undefined>> {
+    enabled = true
+    return getAccounts().catch((error: any) => {
+      // If the error.message is an object, then the error message originated from keepkey
+      // So we grab the string and rethrow so the walletcheck can capture and display
+      if (error?.message?.message) {
+        throw new Error(error?.message?.message)
+      }
+      return []
+    })
+  }
+
+  function disconnect() {
+    dPath = ''
+    addressToPath = new Map()
+    enabled = false
+    provider.stop()
+  }
+
+  async function setPath(path: string, custom?: boolean): Promise<boolean> {
+    if (!isValidPath(path)) {
+      return false
+    }
+
+    if (path !== dPath) {
+      // clear any existing addresses if different path
+      addressToPath = new Map()
+    }
+
+    if (custom) {
+      try {
+        // Convert the path to the addressNList which is what is used by keepkey for signing
+        const addressNList = bip32ToAddressNList(path)
+        const address = await getAddress(addressNList, path)
+        addressToPath.set(address, addressNList)
```

Comment:
Keepkey uses an array which represents the string dPath. Instead of converting back and forth between the two, internally we map the address to the addressNList